### ISSUE TITLE
Test functions with a lifetime parameter that return `Result`s produce incorrect error message

### ISSUE THREAD

**Issue by sanmai-NL** [rust-lang/rust#55228]
This works:
```rust
#[test]
fn unit_1() -> Result<(), &'static str>
{
}
```

But this doesn’t:
```rust
#[test]
fn unit_1<'a>() -> Result<(), &'a str>
{
}
```

```
error: functions used as tests must have signature fn() -> ()
   --> crate/test.rs:162:5
    |
162 | /     fn unit_1<'a>() -> Result<(), &'a str>
```

## Meta

```
rustc 1.31.0-nightly (46880f41b 2018-10-15)
binary: rustc
commit-hash: 46880f41b7aeb897b8245474196bba9dc11f0e88
commit-date: 2018-10-15
host: x86_64-unknown-linux-gnu
release: 1.31.0-nightly
LLVM version: 8.0
```

**Comment by fmease** [comment]
This is now permitted.
